[errors] Show better errors when user script errors out and fix poetry plugin

mikeland73 opened the pull request:
* When a user script errors out, show the error code (e.g. `Error: exit status 1`) This helps in cases when stderr is empty (which the previous `--quiet` was causing)
* Improve poetry plugin by removing `--quiet` flag and adding more required into in readme.

Possible future improements:
* Better error when python is not installed
* Better error when pyproject is missing
* Allow custom pyproject locations

## Summary

## How was it tested?

<img width="352" alt="image" src="https://github.com/jetpack-io/devbox/assets/544948/db93da90-d244-4a0b-bf8f-5ab5812464b7">

New implementation

<img width="381" alt="image" src="https://github.com/jetpack-io/devbox/assets/544948/e2f73145-5e24-4be9-bf5e-1299a2e72c83">

savil commented:
I don't understand the Test Plan. Why is `devbox run echo 5` complaining about python?

savil commented on internal/boxcli/midcobra/midcobra.go:
```diff
@@ -81,6 +81,7 @@ func (ex *midcobraExecutable) Execute(ctx context.Context, args []string) int {
 		var exitErr *exec.ExitError
 		var userExecErr *usererr.ExitError
 		if errors.As(err, &userExecErr) {
+			ux.Ferror(ex.cmd.ErrOrStderr(), userExecErr.Error()+"\n")
```
Suggestion. Also print the Stderr since it may have other info.

We do similar in midcobra/debug.go: https://github.com/jetpack-io/devbox/blob/42d2f713fde840558faeb9d404b8f5e9e2f697dd/internal/boxcli/midcobra/debug.go#L67

mikeland73 commented:
> I don't understand the Test Plan. Why is devbox run echo 5 complaining about python?

Sorry, I should have been more specific. I added `poetry` and then ran the echo line. The error you see is poetry complaining python is not installed. The error is not visible because `--quiet` was removed. The next line (Error exit status 1) is now visible due to the new code that prints the ExitError

mikeland73 commented:
@savil I changed the implementation a bit. Instead of printing error in `midcobra` which is wrong, we print it in the debug middleware.

I added code to ensure telemetry does not log user errors.

savil reviewed (approved):
coolio
